kore/test/Test/Kore/Step.hs

```diff
@@ -516,7 +372,108 @@ runSteps metadataTools stepLimit configuration axioms =
             simplifier
             Map.empty
         )
-        (Limit.replicate stepLimit $ allRewrites axioms)
+        (repeat $ allRewrites axioms)
         (configuration, mempty)
   where
     simplifier = Simplifier.create metadataTools Map.empty
+
+
+----------
+------------  Working toward replacing much of the above
+-----------
+
+-- test_rewriteVariable :: TestTree
+-- test_rewriteVariable =
+--     testGroup "variable rewriting"
+--     [ applyStrategy                  "identity rewrite"
+--         (Start $ var "var")
+--         [Axiom $ var "x" `rewritesTo` var "x"]
+--         (Expect $ var "var")
+--     ]
+
+
+
+test_rewriteConstructorApplication :: TestTree
+test_rewriteConstructorApplication =
+    applyStrategy                              "rewrite: constructor application"
+        ( Start $ pat "c1" ["var"])
+        [ Axiom $ pat "c1" ["x1"] `rewritesTo` pat "c2" ["x1"]
+        , Axiom $ pat "c2" ["x2"] `rewritesTo` pat "c3" ["x2"]
+        ]
+        ( Expect $                             pat "c3" ["var"])
+      where
+        pat = applyConstructorToVariables
+
+            {- API -}
+
+applyStrategy
+    :: HasCallStack
+    => TestName -> Start -> [Axiom] -> Expect -> TestTree
+applyStrategy testName start axioms expected =
+    strategyTestCase (start, axioms) expected testName
+
+
+-- API Helpers
+
+strategyTestCase
+    :: HasCallStack
+    => (Start, [Axiom]) -> Expect -> TestName -> TestTree
+strategyTestCase (start, axioms) expected testName =
+    testCase testName $
+        takeSteps (start, axioms) >>= compareTo expected
+
+takeSteps :: (Start, [Axiom]) -> IO (Actual, Proof)
+takeSteps (Start start, axioms) =
+    runSteps
+        mockMetadataTools
+        (pure start)
+        (unwrap <$> axioms)
+      where
+        unwrap (Axiom a) = a    -- coerce?
+
+compareTo
+    :: HasCallStack
+    => Expect -> (Actual, Proof) -> IO ()
+compareTo (Expect expected) (actual, _ignoredProof) =
+    assertEqualWithExplanation "" (pure expected) actual
+
+-- Builders
+
+-- | Create a function pattern from a function name and list of argnames.
+applyConstructorToVariables :: Text -> [Text] -> TestPattern
+applyConstructorToVariables name arguments =
+    mkApp patternMetaSort symbol $ fmap var arguments
+  where
+    symbol = SymbolOrAlias -- can this be more abstact?
+        { symbolOrAliasConstructor = Id name AstLocationTest
+        , symbolOrAliasParams = []
+        }
+
+-- | Do the busywork of converting a name into a variable pattern.
+var :: Text -> TestPattern
+var name =
+    mkVar $ (Variable (testId name) mempty) patternMetaSort
```

Comment:
Because the sort doesn't really matter for these tests, and because `patternMetaSort` will go away, I would suggest creating an arbitrary sort for this module's tests.

Reply:
Done.